art/estimators/scikitlearn.py

```diff
@@ -26,3 +26,18 @@ class ScikitlearnEstimator(BaseEstimator):
     """
     Estimator class for scikit-learn models.
     """
+
+    def clone_for_refitting(self) -> 'ScikitlearnEstimator':  # lgtm [py/inheritance/incorrect-overridden-signature]
+        """
+        Create a copy of the estimator that can be refit from scratch.
+
+        :return: new estimator
+        """
+        raise NotImplementedError
+
+    def reset(self) -> None:
+        """
+        Resets the weights of the estimator so that it can be refit from scratch.
+
+        """
+        raise NotImplementedError
```

Comment:
I think we should not add these lines as they remove the requirement from classes inheriting from this class of implementing these new methods.